pymongo/ssl_support.py

```diff
@@ -65,10 +80,15 @@ def get_ssl_context(
         allow_invalid_certificates: bool,
         allow_invalid_hostnames: bool,
         disable_ocsp_endpoint_check: bool,
-    ) -> _ssl.SSLContext:
+        is_sync: bool,
+    ) -> Union[_pyssl.SSLContext, _ssl.SSLContext]:  # type: ignore[name-defined]
         """Create and return an SSLContext object."""
+        if is_sync and HAVE_PYSSL:
+            ssl_in_use: types.ModuleType = _pyssl
+        else:
+            ssl_in_use = _ssl
```

Comment:
Can we rename `ssl_in_use` -> `_ssl` that way there's less code churn.

Reply:
I don't think so? because _ssl is referring to stdlib ssl and i don't think we'd want it to be replaced by pyopenssl just because pyopenssl is installed?

Reply:
It's a local variable assignment so what's the issue?
```python
        if is_sync and HAVE_PYSSL:
            _ssl: types.ModuleType = _pyssl
        else:
            _ssl = _ssl
```

Reply:
Ah, no you're right -- I've been conditioned to think that shadowing a global var with a local var is generally bad for code readability(?) so my brain forgot that it was something we could do.

Reply:
Er, am i doing it incorrectly? I'm seeing the error "UnboundLocalError: local variable '_ssl' referenced before assignment"

Reply:
Uh, I swore I've done this before. I guess not. The only way around that is like this:
```python
        if is_sync and HAVE_PYSSL:
            _ssl: types.ModuleType = _pyssl
        else:
            _ssl = globals()["_ssl"]
```

Or by renaming the global "_ssl".

Reply:
I actually kinda wanted to rename the global `_ssl` except i think there were some other modules that imported `_ssl` from `ssl_support` and then i wasn't sure if i was allowed to change it...

Reply:
You know what, I think it's simpler to go back to `ssl_in_use` or just `ssl`?

Reply:
All good, went with `ssl` because its shorter.